src/pydantic_zarr/v2.py

```diff
@@ -318,14 +333,14 @@ def from_zarr(cls, array: zarr.Array) -> Self:
             fill_value=array.dtype.type(array.fill_value).tolist(),
             order=array.order,
             filters=array.filters,
-            dimension_separator=array._dimension_separator,
-            compressor=array.compressor,
+            dimension_separator=array.metadata.dimension_separator,
+            compressor=array.compressors[0].get_config(),
```

Comment:
I believe this will fail if the array has no compressors

Reply:
It did indeed, but should be fixed now.